### DISCUSSION TITLE
reqHistoricalData + keepUpToDate

### DISCUSSION THREAD

**Discussion by kaidaniel82** [erdewit/ib_insync#685] [Q&A]
Is this still True:

"reqHistoricalData + keepUpToDate will, at the moment of writing, leave the whole API inoperable after a network interruption."

**Comment by kkotsche1** [comment]
I am fighting with this issue as well. 

I am using the keepUpToDate flag to run my trading logic where I am checking for certain conditions on each new bar.

For ages now I have been attempting to deal with the case where Interactive Brokers loses connection, leading to interruption of the bar stream and subsequent death of my programs ability to do anything. However, I am unable to figure out any way in which I can get this thing spun back up after a loss of connection to restart the stream and continue on with the trading logic. 

Considering jumping ship either to a different wrapper package or leaving TWS all together as this is just incredibly frustrating.
